consensus/miner_signalled_consensus.md

```diff
@@ -0,0 +1,196 @@
+# Miner Signalled Consensus Upgrade
+
+## Simple Summary
+
+This document describes the proposal of how nodes in the peer-to-peer network
+may activate a new hard-forking consensus protocol based on miner signalling.
+
+## Motivation
+
+Activating a hard-forking new consensus protocol based on miner signalling
+prevents diverging the block chain in case a large fraction of the miners have
+not upgraded their nodes to a version able to follow the new consensus protocol.
+
+## Specification
+
+### Overview
+
+On a node configured for a miner-signalled consensus upgrade,
+per chain fork,
+the new consensus protocol becomes active at a determined
+height if at least a determined majority of the key blocks in a determined
+"signalling" block interval in such chain fork contains a determined value in
+the header field `info`.
+
+The "signalling" block interval has a determined number of confirmations before
+the determined height at which the new consensus protocol may become
+active. This is in order to prevent the node from having to aggregate the
+results of the "signalling" block interval in a too short time interval.
```

Comment:
And also for preventing a user seeing the new protocol active after the fork height to stop node and configure unconditionally the new consensus protocol, and then restarting node (that had other chain fork X) and that chain fork gets prolonged.